mrp_project/models/mrp_production.py

```diff
@@ -63,21 +79,35 @@ def _prepare_production_task(self, production):
     def action_in_production(self):
         """
         Look for tasks related to the current MO without a workorder
+        and create if create_tasks is set and no results where found.
         """
-        task_obj = self.env['project.task']
-        for record in self:
-            task_domain = [('mrp_production_id', '=', record.id),
-                           ('workorder', '=', False)]
-            tasks = task_obj.search(task_domain)
-            if not tasks:
-                task_obj.create(self._prepare_production_task(record))
+        if not self.project_id:
+            raise exceptions.ValidationError(_(
+                "To create a task per work order, a project must be selected. "
```

Comment:
This is the cause of my rejection.
You don't need to create a task per work order. 
One single task will be created for the MO.
If you need creating one task by WO, you just need to install the module mrp_operations_project and set, one single worker for each routing step.

To create a task related to each WO, is not needed if there is no workers on it.